Marlin/configuration_store.cpp

```diff
@@ -324,9 +344,9 @@ void Config_StoreSettings()  {
     EEPROM_WRITE_VAR(i, dummy);
   }
 
-  char ver2[4] = EEPROM_VERSION;
   int j = EEPROM_OFFSET;
-  EEPROM_WRITE_VAR(j, ver2); // validate data
+  EEPROM_WRITE_VAR(j, version);
+  EEPROM_WRITE_VAR(j, eeprom_checksum);
```

Comment:
Aha, I see what you're saying: Notice that if it writes the checksum directly from the "scratch" variable, it will be changing as it's being written! We need to clear the checksum at the right starting point, grab and store it at the right ending point, and write the stored checksum, not the "scratch" checksum.